Library/Homebrew/bottle_checksum.rb

```diff
@@ -0,0 +1,19 @@
+# typed: true
+# frozen_string_literal: true
+
+require "checksum"
+
+# A formula's bottle block checksum.
+#
+# @api private
+class BottleChecksum < Checksum
+  extend Forwardable
+
+  attr_reader :tag, :cellar
+
+  def initialize(hexdigest, tag: nil, cellar: nil)
+    super(Checksum::TYPES.first, hexdigest)
```

Comment:
I don't think `.first` is a good idea here. At this point it's probably worth refactoring (in a separate PR) all uses of `Checksum::TYPES` and probably even `.hash_type` to always assume `sha256` or to add tests/methods for e.g. `sha512` to warrant this being dynamic.

Reply:
This is now simplified with the previous sha256 refactoring.